enumerables.rb

```diff
@@ -0,0 +1,252 @@
+      true
+    # Elsif for Class argument query
+    elsif type.is_a?(Class) and type != Numeric and !type.nil?
+      classflag = true
+      firstclass = self[0].class.name
+      my_each do |m|
+        classflag = false if m.class.name != firstclass
+      end
+      classflag = false if empty?
+      classflag
+    # Elsif for Numeric superclass query
+    elsif type == Numeric
+      classflag = true
+      my_each do |m|
+        if m.class.superclass.name != 'Numeric'
+          classflag = false
+        elsif empty?
+          classflag = false
+        end
+      end
+      classflag
+    elsif type != Numeric and type.is_a?(Numeric)
+      my_each do |m|
+        return false if m != type
+      end
+      true
+    elsif !block_given? and type.nil?
+      my_each do |m|
+        return false if m.nil? || m == false
+      end
+      true
+    end
+  end
+
+  # my_any? Method
+
+  def my_any?(cond = nil)
+    classflag = false
+    if !cond.nil?
+      if cond.is_a?(Class)
+        my_each do |e|
+          if e.is_a?(cond)
+            classflag = true
+            break
+          end
+        end
+      elsif cond == Numeric
+        my_each do |e|
+          if e.is_a?(Numeric)
+            classflag = true
+            break
+          end
+        end
+      elsif cond != Class and cond.is_a?(Regexp)
+        my_each do |e|
+          if e.is_a?(String)
+            if e.match?(cond)
+              classflag = true
+              break
+            end
+          end
+        end
+      elsif !cond.is_a?(Numeric) and !!cond == cond
+        my_each do |e|
+          if e == cond
+            classflag = true
+            break
+          end
+        end
+      elsif cond.is_a?(Numeric) || cond.is_a?(String)
+        my_each do |e|
+          if cond == e
+            classflag = true
+            break
+          end
+        end
+      end
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = true
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      my_each do |m|
+        return true if m == true
+      end
+      false
+    else
+      false
+    end
+    classflag
+  end
+
+  # my_none Simplest Approach
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
+
+  # my_count Method
+  def my_count(cond = nil)
+    counter = 0
+    if block_given?
+      my_each do |f|
+        counter += 1 if yield f
+      end
+      counter
+    elsif cond.nil?
+      to_a.length
+    elsif !cond.nil?
+      my_each do |e|
+        counter += 1 if cond == e
+      end
+      counter
+    end
+  end
+
+  # my_map method
+  def my_map(pro = nil)
+    new_array = []
+    if pro.is_a?(Proc)
+      my_each do |e|
+        new_array.push(pro.call(e))
+      end
+      new_array
+    elsif block_given?
+      my_each do |e|
+        new_array.push(yield e)
+      end
+      new_array
+    else
+      new_array = self
+    end
+    new_array.to_enum
+  end
```

Comment:
### Variables used: 
```Ruby
  array = Array.new(100) { rand(0...9) } 
  range = Range.new(5, 50)
  my_proc = proc { |num| num > 10 }
  block = |num| num < (0 + 9) / 2 
```

- [ ] #my_map is returning Enumerator arrays instead of normal arrays

   

```Ruby

  array.my_map { |num| num < 10 }
  # and
  array.my_map(&block)
  # and
  range.my_map(&block)
  # and
  array.my_map(&my_proc
  # and
  array.my_map(my_proc
  
  #  it should return the same result it gives out now but not as an Enumerator

  # For Example, the first method in this code block does this: 
  # got  #<Enumerator: [true, true, true(...)true, true, true, true]
  # but it should be [true, true, true(...)true, true, true, true]

```

don't return everything `to_enum`, only return if no block is given.